src/prefect/tasks/gcp/bigquery.py

```diff
@@ -403,6 +407,150 @@ def run(
+            containing a JSON representation of your Google Application credentials
+        - **kwargs (optional): additional kwargs to pass to the `Task` constructor
+    """
+
+    def __init__(
+        self,
+        file: Union[str, Path] = None,
+        rewind: bool = False,
+        size: int = None,
+        num_retries=_DEFAULT_NUM_RETRIES,
+        dataset_id: str = None,
+        table: str = None,
+        project: str = None,
+        schema: List[bigquery.SchemaField] = None,
+        location: str = "US",
+        credentials_secret: str = None,
+        **kwargs,
+    ):
+        self.file = file
+        self.rewind = rewind
+        self.size = size
+        self.num_retries = num_retries
+        self.dataset_id = dataset_id
+        self.table = table
+        self.project = project
+        self.schema = schema
+        self.location = location
+        self.credentials_secret = credentials_secret
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs(
+        "file",
+        "rewind",
+        "size",
+        "num_retries",
+        "dataset_id",
+        "table",
+        "project",
+        "location",
+        "credentials_secret",
+    )
+    def run(
+        self,
+        file: Union[str, Path] = None,
+        rewind: bool = False,
+        size=None,
+        num_retries=_DEFAULT_NUM_RETRIES,
+        dataset_id: str = None,
+        table: str = None,
+        project: str = None,
+        schema: List[bigquery.SchemaField] = None,
+        location: str = "US",
+        credentials: dict = None,
+        credentials_secret: str = None,
+        **kwargs,
+    ):
+        """
+        Run method for this Task.  Invoked by _calling_ this Task within a Flow context, after initialization.
+
+        Args:
+            - file (Union[str, path-liike object], optional): A string or path-like object of the file to be loaded
+            - dataset_id (str, optional): the id of a destination dataset to write the
+                records to; if not provided here, will default to the one provided at initialization
+            - table (str, optional): the name of a destination table to write the
+                records to; if not provided here, will default to the one provided at initialization
+            - project (str, optional): the project to initialize the BigQuery Client with; if not provided,
+                will default to the one inferred from your credentials
+            - schema (List[bigquery.SchemaField], optional): the schema to use when creating the table
+            - location (str, optional): location of the dataset that will be written to; defaults to "US"
+            - credentials (dict, optional): a JSON document containing Google Cloud credentials.
+                You should provide these at runtime with an upstream Secret task.
+            - credentials_secret (str, optional, DEPRECATED): the name of the Prefect Secret
```

Comment:
Since this type of secret is being deprecated I think this kwarg should be removed and only `credentials` should be used

Reply:
done